compiler/p/codegen/PPCInstruction.cpp

```diff
@@ -517,7 +517,7 @@ void TR::PPCTrg1Src1Instruction::assignRegisters(TR_RegisterKinds kindToBeAssign
    TR_RegisterKinds  kindOfRegister = sourceVirtual->getKind();
    bool              excludeGPR0 = false;
 
-   if (getOpCodeValue()==TR::InstOpCode::addi || getOpCodeValue()==TR::InstOpCode::addis || getOpCodeValue()==TR::InstOpCode::addi2)
+   if (getOpCodeValue()==TR::InstOpCode::addi || getOpCodeValue()==TR::InstOpCode::addis || getOpCodeValue()==TR::InstOpCode::addi2 || getOpCodeValue()==TR::InstOpCode::lxvl)
       excludeGPR0 = true;
```

Comment:
this testing list just looks alarmingly long and unamicable to maintain (see there are a few places doing this), and it will grow (e.g.  isel).   should think about how to improve this (e.g.  opcode property testing  or  specific FORMAT implication).

Reply:
Agreed. Generally speaking, the `assignRegisters` functions are in desperate need of refactoring. Unfortunately, I can only focus on refactoring a limited set of places in the codegen at any given time, so I haven't had the time to go through and improve these yet. I've opened #5297 to track this particular issue so we don't forget that this should be done.